homeassistant/components/wled/__init__.py

```diff
@@ -125,9 +119,11 @@ async def _async_update_data(self) -> WLEDDevice:
             raise UpdateFailed(f"Invalid response from API: {error}") from error
 
 
-class WLEDEntity(CoordinatorEntity):
+class WLEDEntity(CoordinatorEntity[WLEDDevice]):
```

Comment:
This change seems incorrect. It extends CoordinatorEntity, however, WLED Device isn't into play here at all yet.

Reply:
CoordinatorEntity was made generic in https://github.com/home-assistant/core/pull/49854 . `WLEDDataUpdateCoordinator` returns a `WLEDDevice`  in [`_async_update_data`](https://github.com/home-assistant/core/blob/dev/homeassistant/components/wled/__init__.py#L120), so I believe this typing is correct.

Reply:
It returns an entity... which is a higher abstraction.

Reply:
Hmm. The coordinator update method definitely [appears to return WLEDDevice](https://github.com/home-assistant/core/blob/dev/homeassistant/components/wled/__init__.py#L120) , so I don't see what other type variable the coordinator could have. Apologies if I'm missing something.

Reply:
Oh right 🤦 Good morning ☕  This is not Home Assistant entities but upstream data. 👍